### ISSUE TITLE
[Question] How to optimize two loss alternately with gradient accumulation?

### ISSUE THREAD

**Issue by hkunzhe** [huggingface/accelerate#1870]
I want to update a model by optimizing two loss alternately with gradient accumulation like this

```python
# Suppose gradient_accumulation is set to 2.
optimizer = optim(unet.parameters())
with accelerator.accumulate(unet):
    outputs = unet(input)
    loss1 = loss_func1(outputs)
    loss1.backward()
    optimizer.step()
    optimizer.zero_grad()

with accelerator.accumulate(unet):
    outputs = unet(input)
    loss2 = loss_func2(outputs)
    loss2.backward()
    optimizer.step()
    optimizer.zero_grad()
```

Is this correct?  It appears from the [documentation](https://huggingface.co/docs/accelerate/usage_guides/gradient_accumulation#converting-it-to-accelerate) that `accelerator.accumulate` will normalize the loss and then backpropagate without updating the gradient until reaching `gradient_accumulation_steps`. My main concern is that the gradients accumulated by two different losses for the same model will affect each other.

Hope to find some help here, thanks in advance.

**Comment by muellerzr** [comment]
This question would be better suited for the forums, as the github is mostly meant for bug reports. Please open a new discussion here: https://discuss.huggingface.co/c/accelerate/18 Thanks!

**Comment by muellerzr** [comment]
We can certainly do that! Nice call :)

**Comment by hkunzhe** [comment]
Any ideas @muellerzr? It seems that the discussion on the forum has not been updated for a long time, and my problem is not completely solved.
